### PR TITLE
fix: forceNewDeployment to be a boolean

### PR DESCRIPTION
**Pull request by paragbhingre**
*This PR fix the Issue [#133](https://github.com/aws-actions/amazon-ecs-deploy-task-definition/issues/133)*

*Description of changes:*
This issue was occurring due to incorrect data type of the forceNewDeployment parameter.
In order to fix this issue need to type caste the value of the forceNewDeployment parameter from string to boolean.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -222,8 +222,10 @@ async function run() {
     if (waitForMinutes > MAX_WAIT_MINUTES) {
       waitForMinutes = MAX_WAIT_MINUTES;
     }
-    const forceNewDeployment = core.getInput('force-new-deployment', { required: false }) || false;
 
+    const forceNewDeployInput = core.getInput('force-new-deployment', { required: false })
```
Comment (allisaurus) [inline index.js:226]: this line appears to be missing a terminal `;`

File: index.test.js [hunk 2]
```diff
@@ -755,7 +755,7 @@ describe('Deploy to ECS', () => {
             cluster: 'cluster-789',
             service: 'service-456',
             taskDefinition: 'task:def:arn',
-            forceNewDeployment: true
+            forceNewDeployment: false
```
Comment (piradeepk) [inline index.test.js:758]: the name of this test is force new deployment, but you're giving it a value of false (shouldn't this be false by default/wouldn't you want to test true?)

File: index.js [hunk 3]
```diff
@@ -222,8 +222,10 @@ async function run() {
...
+    const forceNewDeployment = forceNewDeployInput != undefined && forceNewDeployInput.toLocaleLowerCase === "true" ? true : false;
```
Comment (piradeepk) [inline index.js:227]: nit since your conditional returns true or false, do you need to explicitly define `? true : false`?

Comment (allisaurus) [inline index.js:227]: let's use `.toLowerCase` instead, since we're not passing in a locale value

File: index.test.js (hunk 2 above)
Comment (allisaurus) [inline index.test.js:758]: this condition is not correct - this test is meant to test the _success_ of passing in `forceNewDeployment`, so we would expect `ecs:update-service` to be called with this param set as `true`. If this is not occurring, either the setup of the test is wrong or the change is not behaving as we expect.

File: index.js (hunk 3 above)
Comment (allisaurus) [reply index.js:227]: also, nit: I'd prefer we wrap the 2 conditions of the ternary operator in parenthesis to make it clear they're being evaluated together to determine the return condition

**Review by allisaurus — CHANGES_REQUESTED** [review]

File: index.js [hunk 4]
```diff
@@ -222,8 +222,10 @@ async function run() {
     if (waitForMinutes > MAX_WAIT_MINUTES) {
       waitForMinutes = MAX_WAIT_MINUTES;
     }
-    const forceNewDeployment = core.getInput('force-new-deployment', { required: false }) || false;
 
+    const forceNewDeployInput = core.getInput('force-new-deployment', { required: false });
+    const forceNewDeployment = forceNewDeployInput != undefined && (forceNewDeployInput.toLowerCase === 'true' || forceNewDeployInput);
```
Comment (piradeepk) [inline index.js:227]: Curious why you're adding the || forceNewDeployInput as a condition here?

**Review by piradeepk — APPROVED** [review]
Ship it! 💯 Thanks for fixing this so quickly @paragbhingre!

**Comment by ltamrazov** [comment]
hi, thank you for this fix! 

@piradeepk When can we expect a release with this revision?

Thank you!

**Pull request merged**